src/PrometheusPushGateway/PushGateway.php

```diff
@@ -4,113 +4,49 @@
 
 namespace PrometheusPushGateway;
 
-use GuzzleHttp\Client;
 use GuzzleHttp\ClientInterface;
-use GuzzleHttp\Exception\GuzzleException;
+use GuzzleHttp\RequestOptions;
 use Prometheus\CollectorRegistry;
-use Prometheus\RenderTextFormat;
-use RuntimeException;
 
-class PushGateway
+final class PushGateway implements PushGatewayInterface
```

Comment:
Making this final is a  breaking  change, we should Not do this

Reply:
Good catch it was an oversight from my part. I will fix it as soon as possible

Reply:
@LKaemmerling I remove the keyword and updated the code since Guzzle\Psr7 v2 is already out and the previous version started deprecated function usage. Both points have been covered.